lib/nerves_hub_link/client.ex

```diff
@@ -174,6 +182,20 @@ defmodule NervesHubLink.Client do
     _ = apply_wrap(mod(), :handle_error, [data])
   end
 
+  @doc """
+  This function is called internally by NervesHubLink to notify clients of disconnects.
```

Comment:
Could you add a short paragraph about why someone would want to use this function?

Also, it seems like it should be an optional callback since I would assume that without other guidance, most people would go with the default. Right?